runtime/commonMain/src/kotlinx/serialization/internal/SerialClassDescImpl.kt

```diff
@@ -28,17 +30,17 @@ open class SerialClassDescImpl @JvmOverloads constructor(
     // don't change lazy mode: KT-32871, KT-32872
     private val indices: Map<String, Int> by lazy { buildIndices() }
 
-    @JvmOverloads
+    @JvmOverloads // TODO protected
```

Comment:
It is called as `public` from generated code, AFAIR

Reply:
Indeed